packages/terra-button/src/Button.jsx

```diff
@@ -80,7 +80,10 @@ const Button = ({
   ...customProps
   }) => {
   const attributes = Object.assign({}, customProps);
-  const buttonText = text ? <span className={styles.text}>{text}</span> : null;
+
+  const iconWithTextStyle = isReversed ? cx(['text-with-icon-reverse']) : cx(['text-with-icon']);
```

Comment:
I think would be less confusing if we renamed it to iconTextStyleClass

Reply:
Assuming the css for the Badge is robust we can just mimic the same css styling and keep this class as text and only make changes to Button.scss

https://github.com/cerner/terra-core/blob/master/packages/terra-badge/src/Badge.scss#L41-L47

Reply:
Ah that is some solid css. I've brought it in and removed the changes to the jsx.